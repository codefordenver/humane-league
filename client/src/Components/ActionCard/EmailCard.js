import React, {Component} from 'react';
import emailLogo from '../../assets/email.png';
import logAction from '../../utils/logAction';
import fetchActionBody from '../../utils/fetchActionBody';


class EmailCard extends Component {
  constructor () {
    super();
    this.state = {
      actionBody: null,
      showContent: false
    }
    this.fetchActionBody = fetchActionBody.bind(this);
  }

  setActionBody = async () => {
    const actionBody = await this.fetchActionBody('email_contents', this.props.action);
    this.setState({ actionBody });
  }

  resetBody = async (newBody) => {
    this.setState({ actionBody: newBody });
  }

  render () {
    const { title, description, to, cc, bcc, subject } = this.props.action;
    const expanded = this.state.actionBody !== null;

    const buttonText = expanded ? 'SEND' : 'EMAIL';
    const buttonOnClick = expanded ? () => logAction('email_actions', this.props.user, this.props.action) : this.setActionBody;
    const targetLink = expanded ? `mailto:${to}?subject=${subject}&body=${this.state.actionBody}` : null;
    const cancelButton = expanded ? <button onClick={() => this.resetBody(null)}>CANCEL</button>: null;
    const showContentButton = expanded ? <button onClick={() => this.setState({ showContent: !this.state.showContent })}>VIEW EMAIL DETAILS</button> : null;
    const textArea = expanded ? <textarea className="body-text" onChange={(e) => this.resetBody(e.target.value)} value={this.state.actionBody}></textarea> : null;

    const emailContent = this.state.showContent ?
      <div className="email-template">
        <p><strong>To:</strong> {to}</p>
        <p><strong>Subject:</strong> {subject}</p>
      </div>
      : null;

    return (
      <div className="ActionCard email-card">
        <h3>{title}</h3>
        <p className="action-description">{description}</p>
        
        {showContentButton}
        {emailContent}
        {textArea}

        <div className="button-holder">
          <a href={targetLink}>
            <button onClick={ buttonOnClick }>{buttonText}<i className="icon-mail"></i></button>
          </a> 
          {cancelButton}
        </div>
      </div>
    );
  }
}

export default EmailCard;