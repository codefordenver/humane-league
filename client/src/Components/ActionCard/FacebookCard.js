/* eslint-disable no-undef */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import logAction from '../../utils/logAction';
import fetchActionBody from '../../utils/fetchActionBody';

class FacebookCard extends Component {
  constructor() {
    super();
    this.state = {
      actionBody: null
    };
    this.fetchActionBody = fetchActionBody.bind(this);
  }

  componentDidMount () {
    if (this.props.user.admin) {
      this.actionCount();
    }
  }

  setActionBody = async () => {
    const actionBody = await this.fetchActionBody('facebook_contents', this.props.action);
    this.setState({ actionBody });
  }

  resetBody = async (newBody) => {
    this.setState({ actionBody: newBody });
  }

  actionCount = async () => {
    const actionLogFetch = await fetch('/api/v1/actions');
    const actionLog = await actionLogFetch.json();
    const actionCount = actionLog.results.filter(actionLog => (actionLog.action_id === this.props.action.id && actionLog.action_type === 'facebook_actions')).length;
    await this.setState({ actionCount });
  }

  completeAction = () => {
    this.props.removeCompleted('facebook', this.props.action);
    logAction('facebook_actions', this.props.user, this.props.action);
  }

  render() {
    const { title, description, target } = this.props.action;

    let buttonText = 'FACEBOOK';
    let buttonOnClick = this.setActionBody;
    let targetLink = null;
    let cancelButton = null;
    let textArea = null;
    let button = <button onClick={ buttonOnClick }>{buttonText}<i className="icon-mail"></i></button>;

    if (this.state.actionBody !== null && !this.props.action.completed) {
      buttonText = 'GO';
      buttonOnClick = this.completeAction;
      targetLink = target;
      cancelButton = <button onClick={() => this.resetBody(null)}>CANCEL</button>;
      textArea = <textarea className="body-text" onChange={(event) => this.resetBody(event.target.value)} value={this.state.actionBody}></textarea>;
    }

    if (this.props.action.completed) {
      button = null;
    }

    if (this.props.user.admin) {
      buttonText = `${this.state.actionCount} people have taken this action!`;
    }

    return (
      <div className="ActionCard facebook-card">
        <h3>{title}</h3>
        <p className="action-description">{description}</p>

        {textArea}

        <div className="button-holder">
          <a target="_blank" href={targetLink}>
            {button}
          </a> 
          {cancelButton}
        </div>
      </div>
    );
  }
}

export default FacebookCard;

FacebookCard.propTypes = {
  user: PropTypes.object,
  action: PropTypes.object,
  removeCompleted: PropTypes.func  
};