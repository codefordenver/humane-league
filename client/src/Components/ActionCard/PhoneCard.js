import React, { Component } from 'react';
import phoneLogo from '../../assets/phone.png';
import logAction from '../../utils/logAction';
import fetchActionBody from '../../utils/fetchActionBody';

class PhoneCard extends Component {
  constructor () {
    super();
    this.state = {
      actionBody: null
    }
    this.fetchActionBody = fetchActionBody.bind(this);
  }

  setActionBody = async () => {
    const actionBody = await this.fetchActionBody('twitter_contents', this.props.action);
    this.setState({ actionBody });
  }

  resetBody = async (newBody) => {
    this.setState({ actionBody: newBody });
  }

  render () {
    const { title, description, phone_number } = this.props.action;
    const expanded = this.state.actionBody !== null;

    const buttonText = expanded ? 'COMPLETED!' : 'CALL';
    const buttonOnClick = expanded ? () => logAction('phone_actions', this.props.user, this.props.action) : this.setActionBody;
    const phoneNumber = expanded ? <p>{`Call ${phone_number}, and then click completed!`}</p> : null;
    const userFeedback = expanded ? <label>We would love to hear how this call went for you:<textarea className="feedback-textarea" placeholder="How did this call go for you?"></textarea></label> : null;
    const cancelButton = expanded ? <button onClick={() => this.resetBody(null)}>CANCEL</button>: null;
    const textArea = expanded ? <label>Here is an optional script for your call: <textarea onChange={(e) => this.resetBody(e.target.value)} value={this.state.actionBody}></textarea></label> : null;

    return (
      <div className="ActionCard phone-card">
        <div className="action-logo-holder">
          <img className="action-logo" src={phoneLogo} alt="Phone Symbol"/>
        </div>
        <div className="action-card-main">
          <h3>{title}</h3>
          <p>{description}</p>

          {phoneNumber}
          {textArea}

          {userFeedback}
          <div className="button-holder">
            <button onClick={ buttonOnClick }>{ buttonText }</button>
            {cancelButton}
          </div>
        </div> 
      </div>
    );
  }
}

export default PhoneCard;