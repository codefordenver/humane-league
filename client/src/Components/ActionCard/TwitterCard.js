import React, { Component } from 'react';
import twitterLogo from '../../assets/twitter.png';
import logAction from '../../utils/logAction';
import fetchActionBody from '../../utils/fetchActionBody';

class TwitterCard extends Component {
  constructor () {
    super();
    this.state = {
      actionBody: null
    }
    this.fetchActionBody = fetchActionBody.bind(this);
  }

  setActionBody = async () => {
    const actionBody = await this.fetchActionBody('twitter_contents', this.props.action);
    this.setState({ actionBody });
  }

  resetBody = async (newBody) => {
    this.setState({ actionBody: newBody });
  }

  render() {
    const { title, description, target } = this.props.action;
    const expanded = this.state.actionBody !== null;

    const buttonText = expanded ? 'GO' : 'TWEET';
    const buttonOnClick = expanded ? logAction('twitter_actions', this.props.user, this.props.action) : this.setActionBody;
    const targetLink = expanded ? `https://twitter.com/intent/tweet?text=${target} ${this.state.actionBody}` : null;
    const cancelButton = expanded ? <button onClick={() => this.resetBody(null)}>CANCEL</button>: null;
    const textArea = expanded ? <textarea onChange={(e) => this.resetBody(e.target.value)} value={this.state.actionBody}></textarea> : null;

    return (
      <div className="ActionCard twitter-card">
        <h3>{title}</h3>
        <p className="action-description">{description}</p>
        {textArea}

        <div className="button-holder">
          <a href={targetLink} target="_blank">
            <button onClick={ buttonOnClick }>{buttonText}<i className="icon-twitter"></i></button>
          </a>
          {cancelButton}
        </div>
      </div>
    );
  }
}

export default TwitterCard;