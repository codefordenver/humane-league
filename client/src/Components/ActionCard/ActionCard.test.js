import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import TwitterCard from './TwitterCard';
import FacebookCard from './FacebookCard';
import EmailCard from './EmailCard';
import PhoneCard from './PhoneCard';

describe('TwitterCard component tests', () => {
  let renderedTwitterCard;
  let mockAction;
  let mockUser;
  let defaultState;
  beforeEach(() => {
    mockAction = {};
    mockUser = {};
    renderedTwitterCard = shallow( <TwitterCard action={mockAction} user={mockUser}/> );
    defaultState = {
      actionBody: null
    }
  })
  it('renders without crashing', () => {
    expect(renderedTwitterCard).toBeDefined();
    expect(renderedTwitterCard.state()).toEqual(defaultState);
  });
  it('matches SnapShot', () => {
    expect(renderedTwitterCard).toMatchSnapshot();
  })
  
})

describe('FacebookCard component tests', () => {
  let renderedFacebookCard;
  let mockAction;
  let mockUser;
  let defaultState;
  beforeEach(() => {
    mockAction = {};
    mockUser = {};
    renderedFacebookCard = shallow( <FacebookCard action={mockAction} user={mockUser}/> );
    defaultState = {
      actionBody: null
    };
  })
  it('renders without crashing', () => {
    expect(renderedFacebookCard).toBeDefined();
    expect(renderedFacebookCard.state()).toEqual(defaultState);
  });
  it('matches SnapShot', () => {
    expect(renderedFacebookCard).toMatchSnapshot();
  });
  
})

describe('EmailCard component tests', () => {
  let renderedEmailCard;
  let mockAction;
  let mockUser;
  let defaultState;
  beforeEach(() => {
    mockAction = {};
    mockUser = {};
    renderedEmailCard = shallow( <EmailCard action={mockAction} user={mockUser}/> );
    defaultState = {
      actionBody: null,
      showContent: false
    };
  });
  it('renders without crashing', () => {
    expect(renderedEmailCard).toBeDefined();
    expect(renderedEmailCard.state()).toEqual(defaultState);
  });
  it('matches SnapShot', () => {
    expect(renderedEmailCard).toMatchSnapshot();
  });
  
})

describe('PhoneCard component tests', () => {
  let renderedPhoneCard;
  let mockAction;
  let mockUser;
  let defaultState;
  beforeEach(() => {
    mockAction = {};
    mockUser = {};
    renderedPhoneCard = shallow( <PhoneCard action={mockAction} user={mockUser}/> );
    defaultState = {
      actionBody: null
    };
  });
  it('renders without crashing', () => {
    expect(renderedPhoneCard).toBeDefined();
    expect(renderedPhoneCard.state()).toEqual(defaultState);
  });
  it('matches SnapShot', () => {
    expect(renderedPhoneCard).toMatchSnapshot();
  });
})