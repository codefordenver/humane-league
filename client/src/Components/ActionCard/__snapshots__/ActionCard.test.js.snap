// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailCard component tests matches SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmailCard
    action={Object {}}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3 />,
        <p
          className="action-description"
/>,
        null,
        null,
        null,
        <div
          className="button-holder"
>
          <a
                    href={null}
          >
                    <button
                              onClick={[Function]}
                    >
                              EMAIL
                              <i
                                        className="icon-mail"
                              />
                    </button>
          </a>
</div>,
      ],
      "className": "ActionCard email-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "action-description",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      null,
      null,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              href={null}
>
              <button
                            onClick={[Function]}
              >
                            EMAIL
                            <i
                                          className="icon-mail"
                            />
              </button>
</a>,
            null,
          ],
          "className": "button-holder",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                onClick={[Function]}
>
                EMAIL
                <i
                                className="icon-mail"
                />
</button>,
              "href": null,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "EMAIL",
                  <i
                    className="icon-mail"
/>,
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "EMAIL",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon-mail",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "button",
            },
            "type": "a",
          },
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3 />,
          <p
            className="action-description"
/>,
          null,
          null,
          null,
          <div
            className="button-holder"
>
            <a
                        href={null}
            >
                        <button
                                    onClick={[Function]}
                        >
                                    EMAIL
                                    <i
                                                className="icon-mail"
                                    />
                        </button>
            </a>
</div>,
        ],
        "className": "ActionCard email-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "action-description",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        null,
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href={null}
>
                <button
                                onClick={[Function]}
                >
                                EMAIL
                                <i
                                                className="icon-mail"
                                />
                </button>
</a>,
              null,
            ],
            "className": "button-holder",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  onClick={[Function]}
>
                  EMAIL
                  <i
                                    className="icon-mail"
                  />
</button>,
                "href": null,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "EMAIL",
                    <i
                      className="icon-mail"
/>,
                  ],
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "EMAIL",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "icon-mail",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "button",
              },
              "type": "a",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`FacebookCard component tests matches SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FacebookCard
    action={Object {}}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3 />,
        <p
          className="action-description"
/>,
        null,
        <div
          className="button-holder"
>
          <a
                    href={null}
                    target="_blank"
          >
                    <button
                              onClick={[Function]}
                    >
                              FACEBOOK
                              <i
                                        className="icon-facebook"
                              />
                    </button>
          </a>
</div>,
      ],
      "className": "ActionCard facebook-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "action-description",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              href={null}
              target="_blank"
>
              <button
                            onClick={[Function]}
              >
                            FACEBOOK
                            <i
                                          className="icon-facebook"
                            />
              </button>
</a>,
            null,
          ],
          "className": "button-holder",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                onClick={[Function]}
>
                FACEBOOK
                <i
                                className="icon-facebook"
                />
</button>,
              "href": null,
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "FACEBOOK",
                  <i
                    className="icon-facebook"
/>,
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "FACEBOOK",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon-facebook",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "button",
            },
            "type": "a",
          },
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3 />,
          <p
            className="action-description"
/>,
          null,
          <div
            className="button-holder"
>
            <a
                        href={null}
                        target="_blank"
            >
                        <button
                                    onClick={[Function]}
                        >
                                    FACEBOOK
                                    <i
                                                className="icon-facebook"
                                    />
                        </button>
            </a>
</div>,
        ],
        "className": "ActionCard facebook-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "action-description",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href={null}
                target="_blank"
>
                <button
                                onClick={[Function]}
                >
                                FACEBOOK
                                <i
                                                className="icon-facebook"
                                />
                </button>
</a>,
              null,
            ],
            "className": "button-holder",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  onClick={[Function]}
>
                  FACEBOOK
                  <i
                                    className="icon-facebook"
                  />
</button>,
                "href": null,
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "FACEBOOK",
                    <i
                      className="icon-facebook"
/>,
                  ],
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "FACEBOOK",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "icon-facebook",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "button",
              },
              "type": "a",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PhoneCard component tests matches SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PhoneCard
    action={Object {}}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3 />,
        <p
          className="action-description"
/>,
        null,
        null,
        null,
        <div
          className="button-holder"
>
          <button
                    onClick={[Function]}
          >
                    CALL
                    <i
                              className="icon-phone"
                    />
          </button>
</div>,
      ],
      "className": "ActionCard phone-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "action-description",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      null,
      null,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
>
              CALL
              <i
                            className="icon-phone"
              />
</button>,
            null,
          ],
          "className": "button-holder",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "CALL",
                <i
                  className="icon-phone"
/>,
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "CALL",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon-phone",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "button",
          },
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3 />,
          <p
            className="action-description"
/>,
          null,
          null,
          null,
          <div
            className="button-holder"
>
            <button
                        onClick={[Function]}
            >
                        CALL
                        <i
                                    className="icon-phone"
                        />
            </button>
</div>,
        ],
        "className": "ActionCard phone-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "action-description",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        null,
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
>
                CALL
                <i
                                className="icon-phone"
                />
</button>,
              null,
            ],
            "className": "button-holder",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "CALL",
                  <i
                    className="icon-phone"
/>,
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "CALL",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon-phone",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "button",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TwitterCard component tests matches SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TwitterCard
    action={Object {}}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3 />,
        <p
          className="action-description"
/>,
        null,
        <div
          className="button-holder"
>
          <a
                    href={null}
                    target="_blank"
          >
                    <button
                              onClick={[Function]}
                    >
                              TWEET
                              <i
                                        className="icon-twitter"
                              />
                    </button>
          </a>
</div>,
      ],
      "className": "ActionCard twitter-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "action-description",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              href={null}
              target="_blank"
>
              <button
                            onClick={[Function]}
              >
                            TWEET
                            <i
                                          className="icon-twitter"
                            />
              </button>
</a>,
            null,
          ],
          "className": "button-holder",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                onClick={[Function]}
>
                TWEET
                <i
                                className="icon-twitter"
                />
</button>,
              "href": null,
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "TWEET",
                  <i
                    className="icon-twitter"
/>,
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "TWEET",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon-twitter",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "button",
            },
            "type": "a",
          },
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3 />,
          <p
            className="action-description"
/>,
          null,
          <div
            className="button-holder"
>
            <a
                        href={null}
                        target="_blank"
            >
                        <button
                                    onClick={[Function]}
                        >
                                    TWEET
                                    <i
                                                className="icon-twitter"
                                    />
                        </button>
            </a>
</div>,
        ],
        "className": "ActionCard twitter-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "action-description",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href={null}
                target="_blank"
>
                <button
                                onClick={[Function]}
                >
                                TWEET
                                <i
                                                className="icon-twitter"
                                />
                </button>
</a>,
              null,
            ],
            "className": "button-holder",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  onClick={[Function]}
>
                  TWEET
                  <i
                                    className="icon-twitter"
                  />
</button>,
                "href": null,
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "TWEET",
                    <i
                      className="icon-twitter"
/>,
                  ],
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "TWEET",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "icon-twitter",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "button",
              },
              "type": "a",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
